store/sqlstore/post_store.go

```diff
@@ -72,42 +99,85 @@ func (s *SqlPostStore) CreateIndexesIfNotExists() {
 	s.CreateFullTextIndexIfNotExists("idx_posts_hashtags_txt", "Posts", "Hashtags")
 }
 
-func (s *SqlPostStore) Save(post *model.Post) (*model.Post, *model.AppError) {
-	if len(post.Id) > 0 {
-		return nil, model.NewAppError("SqlPostStore.Save", "store.sql_post.save.existing.app_error", nil, "id="+post.Id, http.StatusBadRequest)
-	}
+func (s *SqlPostStore) SaveMultiple(posts []*model.Post) ([]*model.Post, *model.AppError) {
+	channelNewPosts := make(map[string]int)
+	maxDateNewPosts := make(map[string]int64)
+	rootIds := make(map[string]int)
+	maxDateRootIds := make(map[string]int64)
+	for _, post := range posts {
+		if len(post.Id) > 0 {
+			return nil, model.NewAppError("SqlPostStore.Save", "store.sql_post.save.existing.app_error", nil, "id="+post.Id, http.StatusBadRequest)
```

Comment:
Should we continue instead of returning ? It's not atomic anyways because this error can happen for the 2nd or 3rd post and we would have saved the 1st one by then.

Reply:
It is atomic because no change is done into the database until we hit the `Insert` statement after the loop. This operation is "all or nothing", you can insert everything correctly, or you can't insert anything at all.